### PR TITLE
[MRG] Fix to info merging.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/3175
I was able to reproduce the described bug with my own data. The problem occurs when trying to merge infos with list of dicts. In my case it was the hpi info that caused the crash. Fixed by making recursive calls to compare the dicts.
Ready for merge if tests pass.

### CODE REVIEW

**Comment by agramfort** [comment]
can you add a tiny test?

**Comment by jaeilepp** [comment]
Test added.

**Comment by agramfort** [comment]
+1 for merge if CIs are happy

ok @Eric89GXL ?

File: mne/io/meas_info.py [hunk 1]
```diff
@@ -1195,8 +1195,12 @@ def _is_equal_dict(dicts):
     is_equal = []
     for d in tests:
         k0, v0 = d[0]
-        is_equal.append(all(np.all(k == k0) and
-                        np.all(v == v0) for k, v in d))
+        if isinstance(v0, list) and isinstance(v0[0], dict):
```
Comment (wmvanvliet) [inline mne/io/meas_info.py]: this will crash if `v0` is an empty list.

**Comment by Eric89GXL** [comment]
Go ahead and merge after addressing @wmvanvliet's point

File: mne/io/meas_info.py (hunk 1 above)
Comment (jaeilepp) [inline mne/io/meas_info.py]: Good catch.

File: mne/io/meas_info.py [hunk 2]
```diff
@@ -1195,7 +1195,8 @@ def _is_equal_dict(dicts):
     is_equal = []
     for d in tests:
         k0, v0 = d[0]
-        if isinstance(v0, list) and isinstance(v0[0], dict):
+        if (isinstance(v0, (list, np.ndarray)) and len(v0) > 0 and
+                isinstance(v0[0], dict)):
             for k, v in d:
```
Comment (Eric89GXL) [inline mne/io/meas_info.py]: this isn't right, iterating over dicts iterates over the keys. Either do `for k in d:` and get the value yourself, or do `for k, v in d.items()`

Comment (jaeilepp) [inline mne/io/meas_info.py]: ``d`` is actually a tuple. The ``dicts`` are unwrapped on the lines above.

**Comment by agramfort** [comment]
thnaks @jaeilepp !

File: mne/io/meas_info.py (hunk 2 above)
Comment (Eric89GXL) [inline mne/io/meas_info.py]: Ahh yes missed that

**Pull request merged**